docs/DeveloperGuide.md

```diff
@@ -0,0 +1,123 @@
+
+
+### System
+`System` contains the classes that interact with the Operating System and external processes.
+ * [`CommandRunner`](/src/main/java/reposense/system/CommandRunner.java) creates processes that executes commands on the terminal. It consists of many *git* commands.
+ * [`LogsManager`](/src/main/java/reposense/system/LogsManager.java) uses the `java.util.logging` package for logging. The `LogsManager` class is used to manage the logging levels and logging destinations. Log messages are output through: `Console` and to a `.log` file.
+
+
+### Model
+`Model` holds the data structures that are commonly used by the different aspects of *RepoSense*.
+ * [`Author`](/src/main/java/reposense/model/Author.java) stores the `GitHub ID` of an author. Any contributions or commits made by the author, using his/her `GitHub ID` or aliases, will be attributed to the same `Author` object. It is used by `AuthorshipReporter` and `CommitsReporter` to attribute the commit and file contributions to the respective authors.
+ * [`CliArguments`](/src/main/java/reposense/model/CliArguments.java) stores the parsed command line arguments supplied by the user. It contains the configuration settings such as the CSV config file to read from, the directory to output the report to, and date range of commits to analyze. These configuration settings are passed into `RepoConfiguration`.
+ * [`RepoConfiguration`](/src/main/java/reposense/model/RepoConfiguration.java) stores the configuration information from the CSV config file for a single repository, which are the repository's orgarization, name, branch, list of authors to analyse, date range to analyze commits and files from `CliArguments`. 
+ These configuration information are used by:
+    - `GitDownloader` to determine which repository to download from and which branch to check out to.
+    - `AuthorshipReporter` and `CommitsReporter` to determine the range of commits and files to analyze.
+    - `ReportGenerator` to determine the directory to output the report.
+
+
+## HTML Dashboard
+
+ ![Dashboard](images/dashboard-architeture.png)
+
+The dashboard contains two main parts: data `JSONs`(generated by `ReportGenerator`), and the static Dashboard template.
+
+ * As shown in the graph above, there will be one summary `JSON`(`summary.json`) containing the summary information of all repositories, and one set of repo detail `JSON`(`commits.json` and `authorship.json`) for each repository, containing the contribution information of each specified author to that repository.
+
+  * The dashboard template is a set of `HTML` + `CSS` + `Javascript`, which is located in `template/`. When *RepoSense* is built using *Gradle*, the content will be zipped into `template/` and placed under `src/main/resources/`, which will then be copied into the `.jar` file. Whenever *RepoSense* generates a new report, the template will be unzipped to the target location along with the generated `JSON` data files.
```

Comment:
Here can probably include a link on how to guide the user to **build** the project. https://github.com/yamidark/RepoSense/blob/a45969ba0bce9ac08a1fbb768b43cb1d5eef5e89/docs/Build.md